File: app/src/main/java/com/shijiben/data/model/EventStatus.kt
package com.shijiben.data.model

enum class EventStatus(val value: Int) {
    NotStarted(0),
    InProgress(1),
    Completed(2);
    companion object {
        fun fromValue(v: Int): EventStatus = entries.firstOrNull { it.value == v } ?: NotStarted
    }
}

File: lib/data/models/event_status.dart
/// 事件状态枚举。
///
/// 0 = 未开始，1 = 进行中，2 = 已完成。
/// 数据库中以 int 存储，通过 [value] 与 [fromValue] 互转。
enum EventStatus {
  notStarted,
  inProgress,
  completed;

  int get value => index;

  static EventStatus fromValue(int value) => EventStatus.values[value];
}
